templates/node/lib/services/service.js.twig

```diff
@@ -11,32 +12,33 @@ class {{ service.name | caseUcfirst }} extends Service {
      *
 {% endif %}
 {% for parameter in method.parameters.all %}
-     * @param {{ parameter.type | typeName }} {{ parameter.name | caseCamel }}
+     * @param {{ '{' }}{{ parameter.type | typeName }}{{ '}' }} {{ parameter.name | caseCamel }}
 {% endfor %}
-     * @throws Exception
-     * @return {}
+     * @throws {{ '{' }}{{ spec.title | caseUcfirst}}Exception}
+     * @returns {% if method.type == 'location' %}{string}{% else %}{Promise}{% endif %}
+
      */
-    async {{ method.name | caseCamel }}({% for parameter in method.parameters.all %}{{ parameter.name | caseCamel }}{{ parameter | paramDefault }}{% if not loop.last %}, {% endif %}{% endfor %}) {
-        let path = '{{ method.path }}'{% for parameter in method.parameters.path %}.replace(new RegExp('{{ '{' }}{{ parameter.name | caseCamel }}{{ '}' }}', 'g'), {{ parameter.name | caseCamel }}){% endfor %};
-        
+    async {{ method.name | caseCamel }}({% for parameter in method.parameters.all %}{{ parameter.name | caseCamel }}{% if not loop.last %}, {% endif %}{% endfor %}) {
+        let path = '{{ method.path }}'{% for parameter in method.parameters.path %}.replace('{{ '{' }}{{ parameter.name | caseCamel }}{{ '}' }}', {{ parameter.name | caseCamel }}){% endfor %};
+        let payload = {};
+{% for parameter in method.parameters.all %}
+
+        if ({{ parameter.name | caseCamel }} !== undefined) {
+            payload['{{ parameter.name }}'] = {{ parameter.name | caseCamel }};
+        }{% if parameter.required %} else {
```

Comment:
Won't we get a Node error if a required param is missing from the method call anyway?

Reply:
There is not such thing as a required parameter for Node.js or Javascript 🙂 

If a param is not used - it is automatically undefined.

![Bildschirmfoto 2021-06-01 um 10 01 51](https://user-images.githubusercontent.com/1759475/120288346-65102b80-c2c0-11eb-903d-a8cf7b987d2a.png)

Reply:
There is not such thing as a required parameter for Node.js or Javascript 🙂

If a param is not used - it is automatically undefined.

<img width="941" alt="Bildschirmfoto 2021-06-01 um 10 05 59" src="https://user-images.githubusercontent.com/1759475/120288941-0ac39a80-c2c1-11eb-8816-f60ee9973c8b.png">